gapic/templates/%namespace/%name_%version/%sub/services/%service/_async_mixins.py.j2

```diff
@@ -91,10 +91,15 @@
 
         # Wrap the RPC method; this adds retry and timeout information,
         # and friendly error handling.
-        rpc = gapic_v1.method_async.wrap_method(
+        TEMP_CLIENT_INFO = gapic_v1.client_info.ClientInfo(
+            gapic_version=DEFAULT_CLIENT_INFO.gapic_version,
+            grpc_version=None,
+            rest_version=google.auth.__version__
+        )
+        rpc = self._client._transport._wrap_method(  # type: ignore
             self._client._transport.get_operation,
             default_timeout=None,
-            client_info=DEFAULT_CLIENT_INFO,
+            client_info=TEMP_CLIENT_INFO,
         )
```

Comment:
We could move this to `_wrapped_methods` similar to PR https://github.com/googleapis/gapic-generator-python/pull/2175.
In that case, the code here would be
```
rpc = self._transport._wrapped_methods[self._transport.get_operation]
```
also see the changes in `gapic/templates/%namespace/%name_%version/%sub/services/%service/transports/base.py.j2` which are needed to update `_prep_wrapped_messages`

WDYT?

Reply:
What I've done here is a hack (I was trying to figure out why the showcase tests were failing) but isn't something we should be doing.

I think what you're suggesting makes sense. We should precompute the wrapped logic for mixins similar to what we do for other methods.

Reply:
I took a look at the changes in #2175. I'd like to point out that we override `_prep_wrapped_messages` for async transports (both rest and grpc). So the relevant logic should go in there as well.

We also want to make sure that the `kind` property is set (to avoid a gRPC code path for a rest call), if available.

We can address this issue separately first and then rebase our work on top of it.

Reply:
Rebasing on top of #2175 sounds good

Reply:
As per https://github.com/googleapis/gapic-generator-python/pull/2175#issuecomment-2369809636, #2175 will be rebased on top of this PR.

Reply:
wrapping logic for mixins has been addressed.